packages/data-context/src/actions/MigrationActions.ts

```diff
@@ -7,12 +9,141 @@ import {
   moveSpecFiles,
   NonStandardMigrationError,
   SpecToMove,
-  supportFilesForMigration,
 } from '../sources'
+import {
+  tryGetDefaultLegacyPluginsFile,
+  supportFilesForMigration,
+  hasSpecFile,
+  getStepsForMigration,
+  getIntegrationFolder,
+  isDefaultTestFiles,
+  getComponentTestFilesGlobs,
+  getComponentFolder,
+  getIntegrationTestFilesGlobs,
+  getSpecPattern,
+} from '../sources/migration'
+
+export function processConfigViaLegacyPlugins (projectRoot: string, legacyConfig: Partial<Cypress.Config>): Promise<Partial<Cypress.Config>> {
+  return new Promise(async (resolve, reject) => {
+    const pluginFile = legacyConfig.pluginsFile ?? await tryGetDefaultLegacyPluginsFile(projectRoot)
```

Comment:
Seeing async functions inside of a promise constructor tends to make me nervous. I believe we would be swallowing any errors that might be thrown from tryGetDefaultLegacyPluginsFile. Can we do something like this instead?

```suggestion
export async function processConfigViaLegacyPlugins (projectRoot: string, legacyConfig: Partial<Cypress.Config>): Promise<Partial<Cypress.Config>> {
  const pluginFile = legacyConfig.pluginsFile ?? await tryGetDefaultLegacyPluginsFile(projectRoot)
  return new Promise((resolve, reject) => {
```

Reply:
done!